setup.py

```diff
@@ -122,5 +124,6 @@ def build_py_modules(basedir: str, excludes: Sequence = ()) -> Sequence:
         "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
         "Programming Language :: Python :: 3.11",
+        "Programming Language :: Python :: 3.12",
```

Comment:
I noticed that https://github.com/gitpython-developers/gitdb/pull/88, which listed Python 3.12 for gitdb, also removed Python 3.7 from the classifiers list and set `python_requires` to have a lower bound of 3.8, on the grounds 3.7 it is end-of-life. I'm not sure if that should be done in GitPython at this time or not, and I have not made either of those changes here, though I'd be pleased to do so on request. (All of the changes in this pull request are compatible with Python 3.7, and there is no need for GitPython and gitdb to support the same version range, since GitPython is not pinned to require the very latest version of gitdb.)